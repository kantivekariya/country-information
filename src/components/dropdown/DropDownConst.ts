export const DROPDOWN_TEST_CONSTANT = {
  dropDownTestId: "dropDownTestId",
  manageClickEventId: "manageClickEventId",
  selectItemId: "selectItemId",
};
