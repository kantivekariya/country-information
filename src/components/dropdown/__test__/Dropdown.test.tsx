import {
  cleanup,
  fireEvent,
  render,
  screen,
} from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import Dropdown from "../Dropdown";
import { DROPDOWN_TEST_CONSTANT } from "../DropDownConst";

afterEach(cleanup);

const dropDownProps = {
  theme: "light",
  title: "",
  regions: ["Europe"],
  handleChange: () => {},
};

describe("Drop down", () => {
  it("renders Drop down", () => {
    render(<Dropdown {...dropDownProps} />, {
      wrapper: MemoryRouter,
    });
    const dropDownElement = screen.getByTestId(
      DROPDOWN_TEST_CONSTANT.dropDownTestId
    );
    expect(dropDownElement).toBeInTheDocument();
    expect(dropDownElement).not.toBeChecked();
  });

  it("should be click on drop down", () => {
    render(<Dropdown {...dropDownProps} />, {
      wrapper: MemoryRouter,
    });
    const dropDownElement = screen.getByTestId(
      DROPDOWN_TEST_CONSTANT.manageClickEventId
    );
    fireEvent.click(dropDownElement);
    expect(dropDownElement).toBeInTheDocument();
    expect(dropDownElement).not.toBeChecked();
  });

  it("should be select drop down item", () => {
    render(<Dropdown {...dropDownProps} />, {
      wrapper: MemoryRouter,
    });
    const dropDownElement = screen.getByTestId(
      DROPDOWN_TEST_CONSTANT.selectItemId
    );
    fireEvent.click(dropDownElement);
    expect(dropDownElement).toBeInTheDocument();
    expect(dropDownElement).not.toBeChecked();
  });

  it("should be matches snapshot", () => {
    const { baseElement } = render(<Dropdown {...dropDownProps} />, {
      wrapper: MemoryRouter,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
