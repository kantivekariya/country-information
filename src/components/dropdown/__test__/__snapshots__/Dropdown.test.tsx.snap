// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop down should be matches snapshot 1`] = `
<body>
  <div>
    <div
      class="dropdown-container"
      data-testid="dropDownTestId"
    >
      <button
        class="dropdown-btn light"
        data-testid="manageClickEventId"
      >
        <span />
        <i
          class="fas fa-chevron-down"
        />
      </button>
      <div
        class="dropdown-menu
				initial
				hide
        
			"
      >
        <button
          class="dropdown-item light"
          data-testid="selectItemId"
          value="Europe"
        >
          Europe
        </button>
      </div>
    </div>
  </div>
</body>
`;
